.bin/phraseapp/translation-builder.rb

```diff
@@ -0,0 +1,88 @@
+class TranslationBuilder
+  def initialize
+    @log = Logger.new(STDOUT, level: Env::DEBUG ? 'DEBUG' : 'INFO')
+    @plugin_dir         = Const::PLUGIN_DIR
+    @plugin_i18n_dirs   = Const::PLUGIN_I18N_DIRS
+    @locale_file_header = Const::LOCALE_FILE_HEADER
+  end
+
+  def build(locale_name, file_basename)
+    @plugin_i18n_dirs.each do |dir|
+      abs_path = File.join(Dir.pwd, dir)
+      Dir[File.join(abs_path, '**', file_basename)].each do |json_path|
+        json_content = File.read(json_path, :encoding => 'utf-8')
+        translations = JSON.parse(json_content)
+
+        php_path = json_path.sub('.json', '.php')
+        php_file = File.open(php_path, 'w:utf-8')
+
+        php_file.puts(@locale_file_header)
+        php_file.puts("$sLangName = '#{locale_name}';")
+        php_file.puts('$aLang = array(')
+        write_translations_to_php(translations, php_file)
+        php_file.puts(');')
+
+        php_file.close
+        @log.info("Built translation file #{php_path}")
+      end
+    end
+  end
+
+  def get_all_keys
+    keys = []
+
+    get_needed_php_files.each do |file_path|
+      keys += extract_keys_from_php_file(file_path)
+    end
+
+    get_needed_tpl_files.each do |file_path|
+      keys += extract_keys_from_tpl_file(file_path)
+    end
+
+    keys.uniq
+  end
+
+  def extract_keys_from_php_file(file_path)
+    file_content = File.read(file_path, :encoding => 'utf-8')
+    keys = file_content.scan(/translate\(['"]([^'"]+)['"]\)/).flatten
```

Comment:
is it save just to scan for our `Helper::translate`. We know that we should use it - but what about others?

Reply:
Well, it's a matter of convention. I'm not aware of any other way to get the keys from our PHP files.